frontend/src/modules/search/components/SearchBox.js

```diff
@@ -330,12 +327,15 @@ export class SearchBoxBase extends React.Component<InternalProps, State> {
     updateSearchInput = (event: SyntheticInputEvent<HTMLInputElement>) => {
         this.setState({
             search: event.currentTarget.value,
-        }, () => this.updateSearchParams());
+        });
     }
 
-    updateSearchParams = debounce(() => {
-        this.update();
-    }, 500)
+    performSearch = (event: SyntheticKeyboardEvent<HTMLInputElement>) => {
```

Comment:
This is a bit confusing as it doesn't, strictly speaking, performs the search. I'd call it "handleKeyDown" and add a comment that says it performs the search on Enter.